### PR TITLE
chore: upgrade deprecated components in demo-app

### LINKED ISSUE #92: Remove deprecated PatternFly components from demo-app
Tracked in #69

### PR DESCRIPTION
**Pull request by jenny-s51**
## What
Closes #92

### CODE REVIEW

**Comment by jeff-phillips-18** [comment]
Dropdown menus no longer close on click outside the menu.

File: packages/module/src/components/contextmenu/ContextMenu.tsx [hunk 1]
```diff
@@ -1,8 +1,7 @@
 import * as React from 'react';
-import {
-	DropdownMenu as DropdownMenuDeprecated,
-	DropdownContext as DropdownContextDeprecated
-} from '@patternfly/react-core/deprecated';
+import { MenuToggle, MenuToggleElement } from '@patternfly/react-core';
+import { DropdownContext as DropdownContextDeprecated } from '@patternfly/react-core/deprecated';
```
Comment (jeff-phillips-18) [inline packages/module/src/components/contextmenu/ContextMenu.tsx:3]: Is there the ability to not use the deprecated DropdownContext?

File: packages/demo-app-ts/src/utils/useTopologyOptions.tsx [hunk 2]
```diff
@@ -74,115 +67,149 @@ export const useTopologyOptions = (
...
-        <DropdownDeprecated
-          position={DropdownPositionDeprecated.right}
-          toggle={<DropdownToggleDeprecated onToggle={() => setLayoutDropdownOpen(!layoutDropdownOpen)}>{layout}</DropdownToggleDeprecated>}
+        <Dropdown
+          toggle={(toggleRef: React.Ref<MenuToggleElement>) => (
+            <MenuToggle ref={toggleRef} onClick={() => setLayoutDropdownOpen(!layoutDropdownOpen)}>
+              {layout}
+            </MenuToggle>
+          )}
           isOpen={layoutDropdownOpen}
```
Comment (jeff-phillips-18) [inline packages/demo-app-ts/src/utils/useTopologyOptions.tsx:76]:
```suggestion
          isOpen={layoutDropdownOpen}
          onOpenChange={(isOpen) => setLayoutDropdownOpen(isOpen)}
```

Comment (jenny-s51) [reply packages/demo-app-ts/src/utils/useTopologyOptions.tsx:76]: Thank you @jeff-phillips-18 , nice catch. Updated

**Review by jeff-phillips-18 — APPROVED** [review]

File: packages/module/src/components/contextmenu/ContextMenu.tsx [hunk 3]
```diff
@@ -31,30 +28,11 @@ const ContextMenu: React.FunctionComponent<ContextMenuProps> = ({
...
-        }}
-      >
-        <div className={css(styles.dropdown, styles.modifiers.expanded)}>
-          <DropdownMenuDeprecated className={css(topologyStyles.topologyContextMenuCDropdownMenu)} autoFocus>
-            {children}
-          </DropdownMenuDeprecated>
-        </div>
-      </DropdownContextDeprecated.Provider>
+      <div className={css(styles.dropdown, styles.modifiers.expanded)}>
+        <Dropdown toggle={() => <></>} className={css(topologyStyles.topologyContextMenuCDropdownMenu)}>
```
Comment (nicolethoen) [inline packages/module/src/components/contextmenu/ContextMenu.tsx:32]: I'm noticing that the old dropdowns in the context menu demo would close after you select an item. IDK if that's a built in behavior or a quirk of the demo. Just checking it was an intended change?

Comment (jeff-phillips-18) [reply packages/module/src/components/contextmenu/ContextMenu.tsx:32]: IMO, they shouldn't close when checking a checkbox so as to allow the user to select multiple w/o reopening. Not sure why the v4 version closed on each selection but I like it better this way.

Comment (jenny-s51) [reply packages/module/src/components/contextmenu/ContextMenu.tsx]: Thank you @nicolethoen , good find. Updated

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**